File: download_images.py

import os
import requests
from urllib.parse import urlparse

def download_images(url_file, save_dir):
    if not os.path.exists(save_dir):
        os.makedirs(save_dir)

    with open(url_file, 'r') as f:
        urls = f.readlines()

    for url in urls:
        url = url.strip()
        try:
            response = requests.get(url, stream=True)
            response.raise_for_status()  # Raise an exception for bad status codes

            # Get the filename from the URL
            parsed_url = urlparse(url)
            filename = os.path.basename(parsed_url.path)
            
            # Save the image
            with open(os.path.join(save_dir, filename), 'wb') as f:
                for chunk in response.iter_content(chunk_size=8192):
                    f.write(chunk)
            print(f"Successfully downloaded {filename}")

        except requests.exceptions.RequestException as e:
            print(f"Failed to download {url}: {e}")

if __name__ == "__main__":
    download_images('image_urls.txt', 'portfolio_images')

File: download_images.js
import https from 'https';
import fs from 'fs';
import path from 'path';
import { URL } from 'url';

const downloadImages = (urlFile, saveDir) => {
    if (!fs.existsSync(saveDir)) {
        fs.mkdirSync(saveDir);
    }

    const urls = fs.readFileSync(urlFile, 'utf-8').split('\n').filter(Boolean);

    urls.forEach((url, index) => {
        const cleanedUrl = url.replace('[/img]', '').trim();
        if (!cleanedUrl) return;

        try {
            const parsedUrl = new URL(cleanedUrl);
            let filename = path.basename(parsedUrl.pathname);

            if (!filename || filename === 'img' || filename === '/') {
                const extension = path.extname(parsedUrl.pathname);
                filename = `image_${index}${extension || '.jpg'}`;
            }

            const savePath = path.join(saveDir, filename);
            const file = fs.createWriteStream(savePath);

            https.get(cleanedUrl, (response) => {
                if (response.statusCode === 200) {
                    response.pipe(file);
                    file.on('finish', () => {
                        file.close();
                        console.log(`Successfully downloaded ${filename}`);
                    });
                } else {
                    console.error(`Failed to download ${cleanedUrl} - status code: ${response.statusCode}`);
                    fs.unlink(savePath, () => {});
                }
            }).on('error', (err) => {
                fs.unlink(savePath, () => {});
                console.error(`Error downloading ${cleanedUrl}: ${err.message}`);
            });
        } catch (error) {
            console.error(`Invalid URL: ${cleanedUrl}`);
        }
    });
};

downloadImages('image_urls.txt', 'portfolio_images');
